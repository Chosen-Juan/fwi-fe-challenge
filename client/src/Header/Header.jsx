import React from 'react';

import './Header.scss';
import { ReactComponent as CloudColor } from './cloud-color.svg';
import { ReactComponent as CloudEffects } from './cloud-effects.svg';
import NewPlayerModal from '../Modal/NewPlayerModal';

const Header = () => (
  <header id="main-header" className="header level is-mobile">
    <div className="level-left">
      <div className="logo">
        <CloudColor className="logo__color" />
        <CloudEffects className="logo__effects" />
      </div>
      <h1 className="title is-size-6-mobile">FWI Poker Challenge</h1>
    </div>
    <NewPlayerModal />
  </header>
);

export default Header;
