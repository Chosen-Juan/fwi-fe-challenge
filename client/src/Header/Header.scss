@import 'variables';

.header {
  background-color: $charcoal;
  box-shadow: 0 4px 5px 0 rgba($sleeping-panther, 0.14),
    0 1px 10px 0 rgba($sleeping-panther, 0.12),
    0 2px 4px -1px rgba($sleeping-panther, 0.4);
  height: 3.5rem;
  position: fixed;
  top: 0;
  width: 100%;
  z-index: 10;

  .title {
    color: $snow;
    margin: 0 1rem;
  }
}

.logo {
  height: 3.125rem;
  margin-left: 1rem;
  position: relative;
  width: 3.125rem;

  &__color {
    fill: $vivid;
  }

  &__color,
  &__effects {
    left: 0;
    position: absolute;
    top: 0;
  }
}
