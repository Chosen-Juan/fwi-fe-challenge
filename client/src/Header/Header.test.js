import React from 'react';
import { shallow } from 'enzyme';

import Header from './Header';

describe('Header', () => {
  const header = shallow(<Header />);

  it('should match snapshot', () => {
    expect(header).toMatchSnapshot();
  });

});
