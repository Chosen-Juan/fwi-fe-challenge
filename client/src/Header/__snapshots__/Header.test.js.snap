// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match snapshot 1`] = `
<header
  className="header level is-mobile"
  id="main-header"
>
  <div
    className="level-left"
  >
    <div
      className="logo"
    >
      <ForwardRef
        className="logo__color"
      />
      <ForwardRef
        className="logo__effects"
      />
    </div>
    <h1
      className="title is-size-6-mobile"
    >
      FWI Poker Challenge
    </h1>
  </div>
  <Connect(NewPlayerModal) />
</header>
`;
